todo_app/app.py

```diff
@@ -1,33 +1,98 @@
-from flask import request
-from flask import Flask, render_template,redirect
-from todo_app.flask_config import Config
 import os
-from todo_app.ViewModel import ViewModel,Item
+
+import requests
+from flask import Flask, redirect, render_template, request
+from flask_login import LoginManager, UserMixin, login_required, login_user, current_user
+
 import todo_app.data.mongo_items as mongo
+from todo_app.flask_config import Config
+from todo_app.ViewModel import Item, ViewModel
 
 
+class User(UserMixin):
+    def __init__(self, id):
+        self.id = id
+        if id == "dani2525":
+            self.role = "writer"
+        else:
+            self.role = "reader"        
+    
 def create_app():
     app = Flask(__name__)
     app.config.from_object(Config())
 
+    app.config['LOGIN_DISABLED'] = os.getenv('LOGIN_DISABLED') == 'True'
+
     @app.route('/' , methods=['GET'])
+    @login_required 
     def alltodoitems():    
         allitems = mongo.getitems()
         todo_view_model = ViewModel(allitems)
         return render_template("index.html", view_model = todo_view_model)
 
 
     @app.route('/createnewcard' , methods=['POST'] )
+    @login_required 
     def createnewcard ():
+        if current_user.role != "writer":
+            return 'sorry, you cannot create a todo card'
```

Comment:
Nice, it would be good to have similar on your `complete_item` function too (unless we're intentionally allowing readers to do that)

Reply:
yes that's more reasonable, done